lua/lean/ft.lua

```diff
@@ -0,0 +1,10 @@
+local M = {}
+
+local lean3 = require("lean.lean3")
+
+function M.detect()
+  if lean3.is_lean3_project() then lean3.init() end
```

Comment:
For me locally it seems like these lines need to be in the opposite order maybe? Probably because tree-sitter re-activates if the filetype is set (which maybe is a reason we should be using setfiletype instead of setting ft directly?)

But see this link: https://asciinema.org/a/3xYEDx2cUIv857hLXgRcZKIbv

The first time I open the file I get lean4 syntax highlighting (the white underlines are how my colorscheme indicates syntax errors), whereas afterwards if I reorder these two lines so that the .ft line is first, I get proper lean3 highlighting.

Reply:
Yep that makes sense. Haven't used treesitter yet, so didn't notice this. Fixed.